kernel/stack.c

```diff
@@ -85,19 +86,19 @@ Z_SYSCALL_HANDLER(k_stack_alloc_init, stack, num_entries)
 
 void k_stack_cleanup(struct k_stack *stack)
 {
-	__ASSERT_NO_MSG(!_waitq_head(&stack->wait_q));
+	__ASSERT_NO_MSG(_waitq_head(&stack->wait_q) == NULL);
 
-	if (stack->flags & K_STACK_FLAG_ALLOC) {
+	if ((stack->flags & K_STACK_FLAG_ALLOC) == K_STACK_FLAG_ALLOC) {
```

Comment:
Wonder if it could just be !NULL, here (to get the exact behavior as before)

Reply:
>(to get the exact behavior as before)

Aren't both the behaviors the same?
I feel this makes the code clearer than using NULL.
Also adding the NULL will violate another misra rule 11.9. which states that NULL is only used as (void *)0.

Reply:
Strictly speaking, they are the same if K_STACK_FLAG_ALLOC is a single-bit-set value, which it is, so I have no objection, as long as these flags will always be single-bit-set flags :)

Reply:
Definitely prefer != 0 here to testing the flag twice.  That's more readable and much less error-prone.

But that said... Can someone check MISRA to see if this is truly a violation?  Bit testing with the & operator is an inherently boolean operation already, even though sure, it has integer type in C.

Reply:
>Definitely prefer != 0 

Is allowed but NULL is not. 
> That's more readable and much less error-prone.

Although i dont agree with this. i am pushing the changes. if I compare with the excat number( i.e 1) it is more clear than having a catchall check like != 0.